contracts/contracts/chains/ethereum/EthereumMessengers.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/Ownable.sol";
+
+import "../../../interfaces/IMessenger.sol";
+import "../../RestrictedCalls.sol";
+import "../../Resolver.sol";
+
+contract EthereumL1Messenger is IMessenger, RestrictedCalls {
+    function callAllowed(address, address) external view returns (bool) {
+        // The call must be delivered by the Resolver
+        // Since we are on the same chain as the Resolver
+        // we should never see an execution of this function.
+        revert("Unexpected state.");
+    }
+
+    // Simply forward the message to the target on L1
+    /* solhint-disable avoid-tx-origin */
+    function sendMessage(address target, bytes calldata message)
+        external
+        restricted(block.chainid)
+    {
+        target.call(message);
+    }
+}
+
+// In order to keep naming consistent between our rollup specific contracts
+// we are using `EthereumL2Messenger` here as a name even if this contract
+// is supposed to be deployed on L1.
+contract EthereumL2Messenger is IMessenger, Ownable, RestrictedCalls {
```

Comment:
you dont need Ownable again here as it is already there with RestrictedCalls

Reply:
I guess. I am wondering if it has some meaning though since it is present in the `ArbitrumL2Messenger` too.

Reply:
@istankovic ?

Reply:
Omitting `Ownable` should result in the same functional behaviour as having it explicitly present as I believe solc does The Right Thing(tm) during type resolution. Specifically, per https://docs.soliditylang.org/en/v0.8.19/contracts.html#multi-inheritance, Solidity uses C3 linearization (similar to Python), but note the order is reverse to the one used by Python: types that are "more-derived" should come later in the sequence.

I have no preference one way or the other, but if you're going to omit `Ownable` from this one, then I think we should also omit it from all other messenger contracts where it is mentioned explicitly.

Reply:
Thanks for sharing this great source of info @istankovic. 
Since we are not explicitly overriding methods of `Ownable` i guess its safe to proceed with omitting it.

Will do the same for `ArbitrumMessengers` as well.